rearrange/tasks.py

```diff
@@ -97,7 +101,50 @@ def __init__(
             successfully_placed=dict(soap_bottle=False, pan=False, knife=False),
         )
 
-        _, gps, cps = self.unshuffle_env.poses
+        ups, gps, cps = self.unshuffle_env.poses
+
+        self.unshuffle_task_spec_hash = md5_hash_str_as_int(
+            canonicaljson.encode_canonical_json(
+                self.unshuffle_env.current_task_spec.__dict__
+            ).decode("utf-8")
+        )
+
+        seeded_rand = random.Random(self.unshuffle_task_spec_hash)
+        self.openable_obj_name_to_openness_iter = {}
+        openable_onames = []
+        priority_onames = []
+        for up, gp in zip(ups, gps):
+            if up["openness"] is not None:
+                openable_onames.append(up["name"])
+                openness_0 = up["openness"]
+                if gp["openness"] == up["openness"]:
+                    o = up["openness"]
+                    ot = OPENNESS_THRESHOLD
+                    a, b, c, d = 0, max(o - ot, 0.0), min(o + ot, 1.0), 1.0
+                    openness_1 = (
+                        seeded_rand.uniform(a, b)
+                        if seeded_rand.random() < (b - a) / ((b - a) + (d - c))
+                        else seeded_rand.uniform(c, d)
+                    )
+                else:
+                    priority_onames.append(up["name"])
+                    openness_1 = gp["openness"]
+
+                self.openable_obj_name_to_openness_iter[up["name"]] = (lambda a, b: iter(
```

Comment:
Not sure what this does tbh.

Reply:
I believe it must be correct, but a comment might make it easier to follow.

Reply:
Commented! And that's exactly correct, just a needlessly complex way to toggle between values :)